components/tag/Tag.razor.cs

```diff
@@ -78,23 +100,38 @@ private static bool IsPresetColor(string color)
             }
 
             bool result = Regex.IsMatch(color, "^(pink|red|yellow|orange|cyan|green|blue|purple|geekblue|magenta|volcano|gold|lime)(-inverse)?$");
-            if (!result) result = Regex.IsMatch(color, "^#([0-9a-fA-F]{6}|[0-9a-fA-F]{3})$");
             return result;
         }
 
         private void UpdateClassMap()
         {
             string prefix = "ant-tag";
             this.ClassMapper.Add(prefix)
-                .If($"{prefix}-has-color", () => !string.IsNullOrEmpty(Color) && !_presetColor)
+                .If($"{prefix}-has-color", () => _isCustomColor)
                 .If($"{prefix}-hidden", () => Visible == false)
-                .GetIf(() => $"{prefix}-{Color}", () => _presetColor)
-                .If($"{prefix}-checkable", () => Mode == "checkable")
+                .GetIf(() => $"{prefix}-{_color}", () => _isPresetColor)
+                .If($"{prefix}-checkable", () => Checkable)
                 .If($"{prefix}-checkable-checked", () => Checked)
                 .If($"{prefix}-rtl", () => RTL)
                 ;
         }
 
+        private string GetStyle() {
```

Comment:
Could you please call `GetStyle` in the setter of `Color` parameter? This avoids having to call it every time the component render.